clinica/utils/inputs.py

```diff
@@ -0,0 +1,223 @@
+
+        raise:
+            Nothing, we prefer returning a string with the problem written in it, so that the rest of Clinica pipelines
+            can handle the problem properly
+
+        Examples: (path are shortened for readability)
+            - You have the full name of a file:
+                File orig_nu.mgz from FreeSurfer of subject sub-ADNI011S4105 session ses-M00 located in mri folder of
+                FreeSurfer output :
+                    clinica_file_reader(['sub-ADNI011S4105'],
+                                        ['ses-M00'],
+                                        caps_directory,
+                                        {'pattern': 'orig_nu.mgz',
+                                         'description': 'freesurfer file orig_nu.mgz',
+                                         'needed_pipeline': 't1-freesurfer'})
+                    gives: ['/caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/sub-ADNI011S4105_ses-M00/mri/orig_nu.mgz']
+
+            - You have a partial name of the file:
+                File sub-ADNI011S4105_ses-M00_task-rest_acq-FDG_pet.nii.gz in BIDS directory. Here, filename depends on
+                subject and session name :
+                     clinica_file_reader(['sub-ADNI011S4105'],
+                                         ['ses-M00'],
+                                         bids_directory,
+                                         {'pattern': '*fdg_pet.nii*',
+                                          'description': 'FDG PET data'})
+                     gives: ['/bids/sub-ADNI011S4105/ses-M00/pet/sub-ADNI011S4105_ses-M00_task-rest_acq-FDG_pet.nii.gz']
+
+            - Tricky example:
+                Get the file rh.white from FreeSurfer:
+                If you try:
+                    clinica_file_reader(['sub-ADNI011S4105'],
+                                        ['ses-M00'],
+                                        caps,
+                                        {'pattern': 'rh.white',
+                                         'description': 'right hemisphere of outter cortical surface.',
+                                         'needed_pipeline': 't1-freesurfer'})
+                        the following error will arise:
+                        * More than 1 file found::
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/fsaverage/surf/rh.white
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/rh.EC_average/surf/rh.white
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/sub-ADNI011S4105_ses-M00/surf/rh.white
+                Correct usage (wanted for pet-surface): pattern string must be 'sub-*_ses-*/surf/rh.white')
+                    It then gives: ['/caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/sub-ADNI011S4105_ses-M00/surf/rh.white']
```

Comment:
Actually, you still need to add `freesurfer_cross_sectional` in the regexp. Indeed, you will have multiple files with the upcoming `t1-freesurfer-longitdudinal`.